### ISSUE TITLE
coreos_production_vagrant_virtualbox.json has broken SSH

### ISSUE THREAD

**Issue by karmux** [coreos/bugs#2472]
# Issue Report #

### Container Linux Version ###

Any recent stable, beta or alpha version.

### Environment ###

- Vagrant 2.12
- VirtualBox 5.2.12.
- https://github.com/pires/kubernetes-vagrant-coreos-cluster
- Box 1: https://stable.release.core-os.net/amd64-usr/current/coreos_production_vagrant.json
- Box 2: https://stable.release.core-os.net/amd64-usr/current/coreos_production_vagrant_virtualbox.json

### Expected Behavior ###

SSH to CoreOS Vagrant box works with both `coreos_production_vagrant_virtualbox.json` and `coreos_production_vagrant.json`.

### Actual Behavior ###

SSH to CoreOS fails when using `coreos_production_vagrant_virtualbox.json`:
```
==> node-1: Waiting for machine to boot. This may take a few minutes...
    node-1: SSH address: 127.0.0.1:2200
    node-1: SSH username: core
    node-1: SSH auth method: private key
    node-1: Warning: Authentication failure. Retrying...
    node-1: Warning: Authentication failure. Retrying...
    node-1: Warning: Authentication failure. Retrying...
```

`--debug` output:
```
D, [2018-07-03T01:03:39.590021 #26375] DEBUG -- net.ssh.authentication.session[2b08759e4c44]: allowed methods: publickey,password,keyboard-interactive
E, [2018-07-03T01:03:39.590057 #26375] ERROR -- net.ssh.authentication.session[2b08759e4c44]: all authorization methods failed (tried none, publickey)

DEBUG ssh: == Net-SSH connection debug-level log END ==
 INFO ssh: SSH not ready: #<Vagrant::Errors::SSHAuthenticationFailed: SSH authentication failed! This is typically caused by the public/private
keypair for the SSH user not being properly set on the guest VM. Please
verify that the guest VM is setup with the proper public key, and that
the private key path for Vagrant is setup properly as well.>
```

With `coreos_production_vagranr.json` SSH is working.

### Reproduction Steps ###

`git clone https://github.com/pires/kubernetes-vagrant-coreos-cluster`
`cd kubernetes-vagrant-coreos-cluster`
`vagrant up`

**Comment by lucab** [comment]
@karmux thanks for the ticket, but you may want to report your troubles to the relevant repository (https://github.com/pires/kubernetes-vagrant-coreos-cluster), as that isn't under coreos umbrella. If you want to boot a ContainerLinux instance under Vagrant, this is the current documented way: https://coreos.com/os/docs/latest/booting-on-vagrant.html

I'm closing this for now, but let us know if the official documentation procedure doesn't work for you.

**Comment by karmux** [comment]
I've updated reproduction steps for https://github.com/coreos/coreos-vagrant.

It's not reproducible with unchanged https://github.com/coreos/coreos-vagrant.git because it uses ignition. Without ignition most CoreOS + Vagrant projects have broken SSH if they fetch CoreOS box using `coreos_production_vagrant_virtualbox.json` not `coreos_production_vagrant.json`.

- `coreos_production_vagrant_virtualbox.json` - Requires ignition for SSH with default private key.
- `coreos_production_vagrant.json` -  SSH with default private key works without ignition.

**Comment by lucab** [comment]
I'm not sure I'm following. If you change the Vagrantfile to strip out part of it, then yes it won't work. [Comments](https://github.com/coreos/coreos-vagrant/blob/62f3b54958aaef3b75d0ebaa254f7303ac2926c1/Vagrantfile#L80) even states what's going on:
> enable ignition (this is always done on virtualbox as this is how the ssh key is added to the system)

**Comment by karmux** [comment]
I just changed Vagrantfile to reveal the difference between `coreos_production_vagrant_virtualbox.json` and `coreos_production_vagrant.json`. When ignition stuff is commented out in `Vagrantfile` but you change `box_url` to `coreos_production_vagrant.json` SSH starts to work again without ignition in VirtualBox.

`coreos_production_vagrant_virtualbox.json` was working fine year or two ago for my custom Vagrant projects and when I recreated these VMs again few days ago I spent a all day to figure out there is difference between these boxes (json files).
